.ci/common.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# Attention, there is no "-x" to avoid problems on Travis
+set -e
+
+# result of command goes to $RUN_JOB
+# 0 = not run (skip)
+# 1 = run (no skip)
+function should_run_job {
+    SKIP_JOB_BY_FILES=$1
+    SKIP_FILES=$2
+    DEBUG=$3
+
+    echo "Initial value of SKIP_JOB_BY_FILES: $SKIP_JOB_BY_FILES"
+    echo "Files to that are allowed to be skipped: $SKIP_FILES"
+
+    if [[ $SKIP_JOB_BY_FILES != 'false' ]]; then
+         ## Identify previous commit to know how much to check
+         ## Check assumes we are only working with 1 commit if we are in master
+         PREVIOUS_COMMIT="HEAD~1"
```

Comment:
This is what fixes the CI problem in master.

Old commit logic said, "Get me the commits between HEAD and origin/master, and use those to identify whats changed".
Master CI currently does NOTHING but just skip. I believe this to be because the difference between HEAD and origin/master in master CI is nothing. They are the same. I see now way right now to determine how many commits were in the PR when they were merged to master, so we are falling back on the old logic of looking at the top most commit.